docs/source/cc_launcher.md

```diff
@@ -0,0 +1,312 @@
+# External Builders and Launchers
+
+Prior to Hyperledger Fabric 2.0, the process used to build and launch
+chaincode was part of the peer implementation and could not be easily
+customized. All chaincode instantiated on the peer would be "built" using
+language specific logic hard-coded in the peer. This build process would
+generate a Docker container image that would be launched to execute chaincode
+that connected as a client to the peer.
+
+This approach limited chaincode implementations to a handful of languages,
+required Docker to be part of the deployment environment, and prevented
+running chaincode as a long running server process.
+
+Starting with Fabric 2.0, External Builders and Launchers address
+these limitations by enabling operators to extend the peer with
+programs that can build, launch, and discover chaincode. To leverage
+this capability you will need to create your own buildpack and then
+modify the peer core.yaml to include a new `externalBuilder`
+configuration element which lets the peer know an external builder is available. The following sections describe the details of this process.
+
+## External Builder Model
+
+Hyperledger Fabric External Builders and Launchers are loosely based on Heroku
+[Buildpacks](https://devcenter.heroku.com/articles/buildpack-api). A buildpack implementation is simply a collection of
+programs or scripts that transform application artifacts into something that
+can run. The buildpack model has been adapted for chaincode packages and
+extended to support chaincode execution and discovery.
+
+### External Builder and Launcher API
+
+An external builder and launcher consists of four programs or scripts:
+
+- `bin/detect`: Determine whether or not this buildpack should be used to
+  build the chaincode package and launch it.
+- `bin/build`: Transform the chaincode package into executable chaincode.
+- `bin/release`: Provide metadata to the peer about the chaincode. (optional)
+- `bin/run`: Run the chaincode.  (optional)
+
```

Comment:
A nit, but maybe this (optional) should be after `bin/release` and `build/run` and before the colon rather than at the end of the line.